activerecord/lib/active_record/persistence.rb

```diff
@@ -1076,7 +1076,7 @@ def verify_readonly_attribute(name)
 
     def _raise_record_not_destroyed
       @_association_destroy_exception ||= nil
-      raise @_association_destroy_exception || RecordNotDestroyed.new("Failed to destroy the record", self)
+      raise @_association_destroy_exception || RecordNotDestroyed.new("Failed to destroy the #{self.class} record", self)
```

Comment:
Maybe we can even make it similar to RecordNotFound error and add the primary key
```suggestion
      key = self.class.primary_key
      raise @_association_destroy_exception || RecordNotDestroyed.new("Failed to destroy #{self.class} with #{key}=#{send(key)}", self)
```

```https://github.com/rails/rails/blob/396b22d3b6e602061f07a96539148bfb38cead9d/activerecord/lib/active_record/core.rb#L290
```